binary-data-formats.qmd

```diff
@@ -0,0 +1,404 @@
+# Binary data formats {#sec-binary-data}
+
+```{r setup}
+#| results: "asis"
+#| echo: false
+source("_common.R")
+status("drafting")
+```
+
+## Introduction
+
+### Prerequisites
+
+This chapter focuses on large data sets, and binary file formats that are well suited to working with them.
+We'll use a data set of item checkouts from Seattle public libraries, available online at [data.seattle.gov/Community/Checkouts-by-Title/tmmm-ytt6](https://data.seattle.gov/Community/Checkouts-by-Title/tmmm-ytt6).
+The data is distributed as a single 9GB csv file, so it may take some time to download.
+For the purposes of this chapter we'll assume that you have a local copy of the file at `data/seattle-library-checkout.csv`.
+
+```{r load-packages}
+#| message: false
+#| warning: false
+library(tidyverse)
+library(arrow)
+library(duckdb)
+library(dbplyr)
+```
+
+## The arrow package
+
+In this chapter we'll rely on Apache Arrow ([arrow.apache.org](https://arrow.apache.org/)), a multi-language toolbox designed for efficient analysis and transport of large data sets.
+The arrow package provides a standard way to use Apache Arrow in R, and allows you to analyze larger-than-memory datasets using familiar dplyr syntax.
+As an additional benefit, arrow is extremely fast: you'll see some examples later in the chapter.
```

Comment:
Would this be a good place to put a direct link to the arrow R package docs too? I know it's "just" two clicks away at arrow.apache.org, but for some folks it might be just disorienting enough to be impactful?